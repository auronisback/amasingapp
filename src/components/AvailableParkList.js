import React, {Component} from 'react';
import ParkStroke from "./ParkStroke";

/**
 * Component managing the set of available parking lots.
 * */
export default class AvailableParkList extends Component {

  /**
   * Strokes appearance.
   * */
  static STROKE_COLOR = '#777';
  static STROKE_OPACITY = 0.5;

  /**
   * Renders the component.
   * @return The react component
   * */
  render() {
    return (
        <div>
          {this._buildAvailableParks()}
        </div>
    );
  }

  /**
   * Creates elements for the available parking lots.
   * */
  _buildAvailableParks = () => {
    let availableParks = [];
    if(this.props.parks)
      this.props.parks.forEach((elem) => {
        availableParks.push(
            <ParkStroke
                key={elem.id}
                start={elem.start}
                end={elem.end}
                color={AvailableParkList.STROKE_COLOR}
                opacity={AvailableParkList.STROKE_OPACITY}
            />);
      });
    return availableParks;
  }
}