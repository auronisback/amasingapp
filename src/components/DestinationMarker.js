import React, {Component} from 'react';
import {SearchWindow} from './SearchWindow';
import {Marker, OverlayView, Circle} from "react-google-maps";

/**
 * Gets the marker's offset for the search window.
 * */
const getPixelPositionOffset = pixelOffset => (width, height) => ({
  x: -(width / 2) + pixelOffset.x,
  y: -(height / 2) + pixelOffset.y
});

/**
 * Default max distance for searches.
 * */
const DEFAULT_MAX_DISTANCE = 100;

/**
 * Component defining the marker for the destination.
 * */
export class DestinationMarker extends Component {

  /**
   * Distance circle appearance.
   * */
  static CIRCLE_FILL_COLOR = '#F00';
  static CIRCLE_STROKE_WEIGHT = 0;

  /**
   * Offset in pixel for the marker.
   * */
  static markerPixelOffset = {
    x: 0,
    y: -32
  };

  /**
   * React Component's state.
   * */
  state = {
    distance: DEFAULT_MAX_DISTANCE,
    dow: new Date().getDay(),
    time: {
      h: new Date().getHours(),
      m: new Date().getMinutes()
    },
    walkWeight: 50
  };

  /**
   *  Rendering method for React Components
   *  @return Component The react component
   * */
  render() {
    // If no destination has been set, nothing will be rendered
    if(!this.props.destination)
      return null;
    // Conditionally rendering the search window
    const overlay = (this.props.searchVisible ?
            <OverlayView
                position={this.props.destination}
                mapPaneName={OverlayView.OVERLAY_MOUSE_TARGET}
                getPixelPositionOffset={getPixelPositionOffset(DestinationMarker.markerPixelOffset)}
                options={{enableEventPropagation: false}}
            >
              <SearchWindow
                  onSubmit={this.onSubmit}
                  className={'SearchOpened'}
                  isLoading={this.props.isLoading}
                  dow={this.state.dow}
                  distance={this.state.distance}
                  time={this.state.time}
                  walkWeight={this.state.walkWeight}
                  onDowChanged={this.onDowUpdate}
                  onTimeChanged={this.onTimeUpdate}
                  onDistanceChanged={this.onDistanceUpdate}
                  onWalkWeightChanged={this.onWalkWeightUpdate}
              />
            </OverlayView> : null
    );
    return (
        <Marker position={this.props.destination}>
          {overlay}
          <Circle
              center={this.props.destination}
              radius={this.state.distance}
              options={{
                fillColor: DestinationMarker.CIRCLE_FILL_COLOR,
                strokeWeight: DestinationMarker.CIRCLE_STROKE_WEIGHT
              }}
          />
        </Marker>
    );
  }

  /**
   * Handler method when dow is changed into search window.
   * @param {int} dow Selected day of the week
   * */
  onDowUpdate = (dow) => {
    this.setState({dow: dow});
  };

  /**
   * Handler method when time is changed.
   * @param {Object} time Time in the search window
   * */
  onTimeUpdate = (time) => {
    this.setState({time: time});
  };

  /**
   * Handler method for changing in the max distance.
   * @param {string} distance Max distance for parks
   * */
  onDistanceUpdate = (distance) => {
    this.setState({distance: +distance > 0 ? +distance : 0});
  };

  /**
   * Handler method for updating the walking weight.
   * @param {number} walkWeight Weight given to walk
   * */
  onWalkWeightUpdate = (walkWeight) => {
    this.setState({walkWeight: walkWeight});
  };

  /**
   * Handles the submitting of search parameters.
   * */
  onSubmit = () => {
    if(this.props.onSearch)
      this.props.onSearch(this.state);
  };


}