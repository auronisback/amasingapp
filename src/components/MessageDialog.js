import React, {Component} from 'react';

const NO_RESULTS = 'Server returned status code ZERO_RESULTS';

export class MessageDialog extends Component {

  /**
   * Constants for error types.
   * */
  static TYPE_MESSAGE = 'info';
  static TYPE_WARNING = 'warning';
  static TYPE_ERROR = 'danger';

  /**
   * Renderizes the component.
   * @return The React Component
   * */
  render() {
    let message = this.getMessage();
    const typeClass = 'alert-' + this.props.type;
    return (
        <div className={'alert ' + typeClass + ' ErrorDialog'}
             onClick={this.props.onClose}
        >
          <h4 className={'alert-heading'}>Warning!</h4>
          <p>{message}</p>
          <span className={'CloseError'}>Click to close</span>
        </div>
    );
  }

  /**
   * Gets the error message from component's properties.
   * @return string The error message in human readable form
   * */
  getMessage = () => {
    switch(this.props.message) {
      case NO_RESULTS:
        return 'The search produced no results.';
      default:
        return this.props.message;
    }
  };

}