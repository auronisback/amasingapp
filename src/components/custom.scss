// Override default variables before the import
$body-bg: #DDD;

// Color palette
$backMain: #4484CE;
$backSecond: #F9CF00;
$foreMain: #F19F4d;
$foreSecond: #D9D9D9;

// Import Bootstrap and its default variables
@import '~bootstrap/scss/bootstrap.scss';

.App {
  height: 100vh;
  width: 100%;
  overflow: hidden;
}

.TopBar {
  background: $backSecond;
  max-width: 100%;
}

.TopBar h1 {
  font-size: 200%;
  margin: 0.6rem 0;
  font-weight: bold;
  color: $backMain;
}

.TopBar ul {
  margin-bottom: 0;
  text-align: right;
}

.TopBar li {
  display: inline-block;
  padding: 2%;
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.TopBar .SearchButton {
  -webkit-border-radius: 0 0.25rem 0.25rem 0;
  -moz-border-radius: 0 0.25rem 0.25rem 0;
  border-radius: 0 0.25rem 0.25rem 0;
}

.MapPanel {
  height: 100vh;
  width: 100%;
}

.DestinationMarker {
  position: relative;
  display: inline-block;
  text-align: center;
  padding: 0.4em;
  width: 200px;
}

.DestinationMarker .Body {
  position: absolute;
}

.DestinationMarker .Body.SearchOpened {
  left: -125px;
  top: -255px;
}

.DestinationMarker .Body.SearchClosed {
  left: -11px;
  top: -30px;
}

.MarkerIcon {
  font-size: 30px;
  display: inline-block;
  text-align: center;
}

.Marker {
  color: $backMain;
}

.ParkMarker {
  margin-top: -35px;
  margin-left: -8px;
}

.SearchPanel {
  display: inline-block;
  padding: 1em;
  margin-bottom: 0.2em;
  background: $backSecond;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  border-radius: 7px;
}

.SearchPanel input, .SearchPanel select {
  margin-bottom: 0.4em;
}

.SearchPanel .WalkingSlider {
  color: $backMain;
  font-size: 12pt;
  margin: 0.6em 0;
}

.SearchPanel .Slider {
  display: inline-block;
  width: 80%;
}

.SearchPanel .Slider .rc-slider-track {
  background: $backMain;
}

.SearchPanel .Slider .rc-slider-handle {
  border-color: $backMain;
}

.SearchPanel .IconWrapper {
  display: inline-block;
  width: 10%;
}

.SearchPanel .IconWrapper:first-child {
  text-align: left;
}

.SearchPanel .IconWrapper:last-child {
  text-align: right;
}

.ErrorDialog {
  cursor: pointer;
  position: fixed;
  bottom: 1em;
  right: 1em;
}

.CloseError {
  font-style: italic;
  font-size: 80%;
}