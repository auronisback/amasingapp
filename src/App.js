import React, {Component} from 'react';
import Geocode from 'react-geocode';
import axios from 'axios';

import './components/custom.scss';
import {TopBar} from './components/TopBar';
import {MapPanel} from './components/MapPanel';
import {DestinationMarker} from './components/DestinationMarker';
import {MessageDialog} from './components/MessageDialog';
import AvailableParkList from "./components/AvailableParkList";
import SuggestedParkList from "./components/SuggestedParkList";

/**
 * Constant for the San Francisco City center.
 * */
const SF_CENTER = {
  lat: 37.773972,
  lng: -122.431297
};

/**
 * App class
 * ControllerView for the app. Manages the creation of views
 * and it is the top component to which view components send
 * user actions and owns the application centralized state.
 * */
class App extends Component {

  /**
   * Message shown when no parks are found.
   * */
  static parksNotFoundMessage = 'No parking lots found with given criteria.';

  /**
   * Application configuration.
   * */
  static configuration = {
    gMapsApiPrefix: 'T4m_9QA5arCySazIA',
    defaultMapCenter: SF_CENTER,
    mapZoom: 13,
    measurementsEndpoint: 'http://18.184.94.248:9090/measurements',
    allParksEndpoint: 'http://18.184.94.248:9090/getParks',
    gMapsApiSuffix: 'E2csRyNvQIoKRBIVi-kRQU',
  };

  /**
   * Application state.
   * */
  state = {
    destination: null, // User destination
    parks: [], // Parking lots found
    availableParks: [], // All parking lots
    searchVisible: true, // Initially the search window is visible
    searching: false // Flag indicating the search is still in progress
  };

  /**
   * Creates the application object.
   * @param {Object} props React Component's properties
   * */
  constructor(props) {
    super(props);
    // Initializing the AjaxWrapper

    // Initializing the Geocoder
    Geocode.setApiKey(App.getAPIKey());
    // Loading all parking in order to show them on the map
    this._loadAllParks();
  }

  /**
   * Renders the application.
   *
   * @output The React component for the application
   * */
  render() {
    // Building the error dialog if needed
    let dialog = null;
    if(this.state.error)
      dialog = (
          <MessageDialog
              message={this.state.error.message}
              type={this.state.error.type}
              onClose={this.onDialogClosed}
          />
      );
    return (
        <div className={'App'}>
          <TopBar
              onReset={this.reset}
              onSearch={this.searchStreetByName}
          />
          <MapPanel
              gMapsApiKey={App.getAPIKey()}
              center={this.state.destination ? this.state.destination : App.configuration.defaultMapCenter}
              zoom={this.state.destination ? App.configuration.destinationMapZoom : App.configuration.mapZoom}
              onDestinationSelection={this.setDestination}
          >
            <AvailableParkList parks={this.state.availableParks}/>
            <DestinationMarker
                destination={this.state.destination}
                searchVisible={this.state.searchVisible}
                onSearch={this.searchParks}
                onSearchClosed={this.onSearchWindowClosed}
                onMarkerClick={this.onDestinationClick}
                isLoading={this.state.searching}
            />
            <SuggestedParkList parks={this.state.parks}/>
          </MapPanel>
          {dialog}
        </div>
    );
  }

  /**
   * Handles the click on the destination marker.
   * */
  onDestinationClick = () => {
    this.setState({searchVisible: true});
  };

  /**
   * Handles the closing of the search window.
   * */
  onSearchWindowClosed = () => {
    this.setState({searchVisible: false});
  };

  /**
   * Sets the destination, if none has been set already.
   *
   * @param {number} lat The destination latitude
   * @param {number} lng The destination longitude
   * */
  setDestination = (lat, lng) => {
    if(this.state.destination == null)
      this.setState({
        destination: {
          lat: lat,
          lng: lng
        }
      });
  };

  /**
   * Resets the destination and found parks.
   * */
  reset = () => {
    this.setState({
      destination: null,
      parks: [],
      searchVisible: true
    });
  };

  /**
   * Search a street by name.
   * */
  searchStreetByName = (streetName) => {
    // Geocoding (adding the city name)
    Geocode.fromAddress(streetName + ' San Francisco').then((response) => {
      // Getting the first element and setting destination
      const {lat, lng} = response.results[0].geometry.location;
      this.setState({
        destination: {
          lat: lat,
          lng: lng
        }
      });
    }, (error) => {
      this._showMessageDialog(error.message, MessageDialog.TYPE_ERROR)
    });
  };

  /**
   * Handler method for closing error window.
   * */
  onDialogClosed = () => {
    // Just setting the error to null
    this.setState({error: null});
  };

  /**
   * Sends the search request for parks. It will
   * asynchronously call the parse response method
   * when data is ready.
   *
   * @param {number} params The search parameters
   * @return Promise The promise that has to be fulfilled
   * */
  searchParks = (params) => {
    this.setState({parks: [], searching: true});
    return this._sendRequest(params);
  };

  /**
   * Sends a request to the endpoint in order to obtain park lots.
   * @param {object} params The search parameters
   * @return Promise The async function
   * */
  _sendRequest(params) {
    console.log("Ehy, I changed something");
    return axios.post(App.configuration.measurementsEndpoint, {
      lat: this.state.destination.lat,
      lon: this.state.destination.lng,
      maxDistance: params.distance,
      dow: params.dow,
      hour: params.time.h,
      minute: params.time.m,
      walkWeight: 1 - params.walkWeight / 100
    }).then((response) => {
      this._parseMeasurementResponse(response.data);
      this.setState({searching: false});
    }).catch((error) => {
      this._showMessageDialog(error.message, MessageDialog.TYPE_ERROR);
      this.setState({searching: false});
    });
  };

  /**
   * Gets the data for a park.
   * @param {number} elem The data for the parking sent by the server
   * @return Object The object with parking information
   * */
  static _getParkData(elem) {
    return {
      id: elem.parkAgent.id,
      start: {
        lat: elem.parkAgent.line.coordinates[0][1],
        lng: elem.parkAgent.line.coordinates[0][0]
      },
      end: {
        lat: elem.parkAgent.line.coordinates[1][1],
        lng: elem.parkAgent.line.coordinates[1][0]
      },
      streetType: elem.parkAgent.highway,
      number: elem.parkAgent.parkingSpaceNumber,
      haveMes: elem.parkAgent.haveMes,
      distance: elem.distance,
      occupancy: elem.meanOccRate,
    };
  };

  /**
   * Sends a request to load all parking lots.
   * */
  _loadAllParks = () => {
    axios.get(App.configuration.allParksEndpoint)
        .then((response) => {
              this._parseAvailableParkResponse(response.data);
            },
            (error) => {
              // Error in the request: showing an error message
              this._showMessageDialog(error.message, MessageDialog.TYPE_ERROR);
            });
  };

  /**
   * Parse the available park data response.
   * @param {Object} data Available park data
   * */
  _parseAvailableParkResponse = (data) => {
    let parks = [];
    data.forEach((elem) => {
      parks.push({
        id: elem.id,
        start: {
          lat: elem.line.coordinates[0][1],
          lng: elem.line.coordinates[0][0]
        },
        end: {
          lat: elem.line.coordinates[1][1],
          lng: elem.line.coordinates[1][0]
        }
      });
    });
    this.setState({availableParks: parks});
  };

  /**
   * Parses the measurements endpoint response.
   * @param {Object} data The response's data
   * */
  _parseMeasurementResponse = (data) => {
    let parks = [];
    data.forEach((elem) => {
      parks.push(App._getParkData(elem));
    });
    if(parks.length === 0) {
      // No parks found: showing an error message
      this._showMessageDialog(App.parksNotFoundMessage, MessageDialog.TYPE_MESSAGE);
    } else {
      // Updating parks
      this.setState({
        parks: parks,
        searchVisible: false
      });
    }
  };

  /**
   * Shows a dialog window in the UI.
   * @param {string} message The message that has to be displayed
   * @param {int} type Type of the dialog
   * */
  _showMessageDialog = (message, type = MessageDialog.TYPE_ERROR) => {
    this.setState({
      error: {
        message: message,
        type: type
      }
    });
  };

  /**
   * Gets the API key.
   * @returns {string} The Google Maps API key.
   * */
  static getAPIKey() {
    return App.configuration.gMapsApiPrefix.split('').reverse().join('') +
        App.configuration.gMapsApiSuffix.split('').reverse().join('');
  };

}

export default App;
