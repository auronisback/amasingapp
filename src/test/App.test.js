import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, {shallow, mount} from 'enzyme';
import App from '../App';
import {MessageDialog} from '../components/MessageDialog';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Configuring adapter
Enzyme.configure({adapter: new Adapter()});

describe('<App>', () => {

  it('Should show the marker if a destination is clicked', () => {
    const app = shallow(<App/>).instance();
    app.onDestinationClick();
    expect(app).toMatchSnapshot();
  });

  it('Should reset marker and destination on demand', () => {
    const app = shallow(<App/>).instance();
    app.reset();
    expect(app).toMatchSnapshot();
  });

  it('Should close the search window', () => {
    const app = shallow(<App/>).instance();
    app.onSearchWindowClosed();
    expect(app).toMatchSnapshot();
  });

  it('Should set or unset the destination', () => {
    const app = shallow(<App/>).instance();
    app.setDestination({lat: 12, lng: 13});
    expect(app).toMatchSnapshot();
    app.setDestination(null);
    expect(app).toMatchSnapshot();
  });

  it('Should show a message dialog', () => {
    const app = shallow(<App/>).instance();
    app._showMessageDialog('Test Message', MessageDialog.TYPE_MESSAGE);
    expect(app).toMatchSnapshot();
    app.onDialogClosed();
    expect(app).toMatchSnapshot();
    app._showMessageDialog('Test Error Message');
    expect(app).toMatchSnapshot();
  });

  it('Should search by street name', () => {
    const app = shallow(<App/>).instance();
    app.searchStreetByName('1st Street');
    expect(app).toMatchSnapshot();
  });

  it('Should show available parkings', () => {
    const app = shallow(<App />).instance();
    const response = [
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4319472943,
              37.7798782941
            ],
            [
              -122.4320478777,
              37.7803731644
            ]
          ]
        },
        "haveMes": 0,
        "highway": "residential",
        "parkingSpaceNumber": 8,
        "id": 415112
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4322279219,
              37.7805954536
            ],
            [
              -122.432372762,
              37.7813095957
            ]
          ]
        },
        "haveMes": 0,
        "highway": "residential",
        "parkingSpaceNumber": 13,
        "id": 415121
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4324572521,
              37.7817290284
            ],
            [
              -122.4325779521,
              37.7823224703
            ]
          ]
        },
        "haveMes": 0,
        "highway": "residential",
        "parkingSpaceNumber": 10,
        "id": 415131
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4323254878,
              37.7817471334
            ],
            [
              -122.432427077,
              37.7822510561
            ]
          ]
        },
        "haveMes": 0,
        "highway": "residential",
        "parkingSpaceNumber": 6,
        "id": 415132
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4330436532,
              37.785290686
            ],
            [
              -122.4331291491,
              37.7857151479
            ]
          ]
        },
        "haveMes": 0,
        "highway": "tertiary",
        "parkingSpaceNumber": 7,
        "id": 415172
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4332428083,
              37.7862804265
            ],
            [
              -122.433371555,
              37.786919131
            ]
          ]
        },
        "haveMes": 0,
        "highway": "tertiary",
        "parkingSpaceNumber": 11,
        "id": 415182
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4337376785,
              37.7880617582
            ],
            [
              -122.4338774894,
              37.7887557835
            ]
          ]
        },
        "haveMes": 0,
        "highway": "tertiary",
        "parkingSpaceNumber": 10,
        "id": 415201
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4342939046,
              37.7908237779
            ],
            [
              -122.4344166164,
              37.7914313015
            ]
          ]
        },
        "haveMes": 0,
        "highway": "tertiary",
        "parkingSpaceNumber": 7,
        "id": 415231
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.4358227721,
              37.7990716154
            ],
            [
              -122.4358801046,
              37.7993562663
            ]
          ]
        },
        "haveMes": 0,
        "highway": "tertiary",
        "parkingSpaceNumber": 5,
        "id": 415322
      },
      {
        "line": {
          "type": "LineString",
          "coordinates": [
            [
              -122.3968769033,
              37.7855286539
            ],
            [
              -122.3978123114,
              37.7847887749
            ]
          ]
        },
        "haveMes": 0,
        "highway": "primary",
        "parkingSpaceNumber": 10,
        "id": 418062
      }
    ];
    app._parseAvailableParkResponse(response);
    expect(app).toMatchSnapshot();
  });

  it('Should display parkings when searched', () => {
    const app = shallow(<App/>).instance();
    const parkResponse = [
      {
        "parkAgent": {
          "line": {
            "type": "LineString",
            "coordinates": [
              [
                -122.3968769033,
                37.7855286539
              ],
              [
                -122.3978123114,
                37.7847887749
              ]
            ]
          },
          "haveMes": 0,
          "highway": "primary",
          "parkingSpaceNumber": 10,
          "id": 418062
        },
        "distance": 172.11040982285127,
        "meanOccRate": 0.5441332389978301
      },
      {
        "parkAgent": {
          "line": {
            "type": "LineString",
            "coordinates": [
              [
                -122.400417421,
                37.7872912895
              ],
              [
                -122.3991661637,
                37.7862874673
              ]
            ]
          },
          "haveMes": 1,
          "highway": "secondary",
          "parkingSpaceNumber": 9,
          "id": 582012
        },
        "distance": 164.2239301886187,
        "meanOccRate": 0.6283333333333334
      },
      {
        "parkAgent": {
          "line": {
            "type": "LineString",
            "coordinates": [
              [
                -122.3988905653,
                37.7836310601
              ],
              [
                -122.3976191913,
                37.7826131563
              ]
            ]
          },
          "haveMes": 1,
          "highway": "primary",
          "parkingSpaceNumber": 24,
          "id": 203032
        },
        "distance": 159.01291189194936,
        "meanOccRate": 0.6691666666666667
      }
    ];
    app._parseMeasurementResponse(parkResponse);
    expect(app).toMatchSnapshot();
  });

  it('Should show a message if no suggested parking has been found', () => {
    const app = shallow(<App />).instance();
    app._parseMeasurementResponse([]);
    expect(app).toMatchSnapshot();
  });
});
