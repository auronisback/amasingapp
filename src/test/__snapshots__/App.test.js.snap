// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> Should close the search window 1`] = `
App {
  "_loadAllParks": [Function],
  "_parseAvailableParkResponse": [Function],
  "_parseMeasurementResponse": [Function],
  "_showMessageDialog": [Function],
  "context": Object {},
  "onDestinationClick": [Function],
  "onDialogClosed": [Function],
  "onSearchWindowClosed": [Function],
  "props": Object {},
  "refs": Object {},
  "reset": [Function],
  "searchParks": [Function],
  "searchStreetByName": [Function],
  "setDestination": [Function],
  "setState": [Function],
  "state": Object {
    "availableParks": Array [],
    "destination": null,
    "parks": Array [],
    "searchVisible": false,
    "searching": false,
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_currentlyRenderingComponent": null,
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <App />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_previousComponentIdentity": null,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="App"
      >
        <TopBar
          onReset={[Function]}
          onSearch={[Function]}
        />
        <MapPanel
          center={
            Object {
              "lat": 37.773972,
              "lng": -122.431297,
            }
          }
          gMapsApiKey="AIzaSyCra5AQ9_m4TUQRk-iVIBRKoIQvNyRsc2E"
          onDestinationSelection={[Function]}
          zoom={13}
        >
          <AvailableParkList
            parks={Array []}
          />
          <DestinationMarker
            destination={null}
            isLoading={false}
            onMarkerClick={[Function]}
            onSearch={[Function]}
            onSearchClosed={[Function]}
            searchVisible={false}
          />
          <SuggestedParkList
            parks={Array []}
          />
        </MapPanel>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;

exports[`<App> Should set or unset the destination 1`] = `
App {
  "_loadAllParks": [Function],
  "_parseAvailableParkResponse": [Function],
  "_parseMeasurementResponse": [Function],
  "_showMessageDialog": [Function],
  "context": Object {},
  "onDestinationClick": [Function],
  "onDialogClosed": [Function],
  "onSearchWindowClosed": [Function],
  "props": Object {},
  "refs": Object {},
  "reset": [Function],
  "searchParks": [Function],
  "searchStreetByName": [Function],
  "setDestination": [Function],
  "setState": [Function],
  "state": Object {
    "availableParks": Array [],
    "destination": Object {
      "lat": Object {
        "lat": 12,
        "lng": 13,
      },
      "lng": undefined,
    },
    "parks": Array [],
    "searchVisible": true,
    "searching": false,
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_currentlyRenderingComponent": null,
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <App />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_previousComponentIdentity": null,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="App"
      >
        <TopBar
          onReset={[Function]}
          onSearch={[Function]}
        />
        <MapPanel
          center={
            Object {
              "lat": Object {
                "lat": 12,
                "lng": 13,
              },
              "lng": undefined,
            }
          }
          gMapsApiKey="AIzaSyCra5AQ9_m4TUQRk-iVIBRKoIQvNyRsc2E"
          onDestinationSelection={[Function]}
        >
          <AvailableParkList
            parks={Array []}
          />
          <DestinationMarker
            destination={
              Object {
                "lat": Object {
                  "lat": 12,
                  "lng": 13,
                },
                "lng": undefined,
              }
            }
            isLoading={false}
            onMarkerClick={[Function]}
            onSearch={[Function]}
            onSearchClosed={[Function]}
            searchVisible={true}
          />
          <SuggestedParkList
            parks={Array []}
          />
        </MapPanel>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;

exports[`<App> Should set or unset the destination 2`] = `
App {
  "_loadAllParks": [Function],
  "_parseAvailableParkResponse": [Function],
  "_parseMeasurementResponse": [Function],
  "_showMessageDialog": [Function],
  "context": Object {},
  "onDestinationClick": [Function],
  "onDialogClosed": [Function],
  "onSearchWindowClosed": [Function],
  "props": Object {},
  "refs": Object {},
  "reset": [Function],
  "searchParks": [Function],
  "searchStreetByName": [Function],
  "setDestination": [Function],
  "setState": [Function],
  "state": Object {
    "availableParks": Array [],
    "destination": Object {
      "lat": Object {
        "lat": 12,
        "lng": 13,
      },
      "lng": undefined,
    },
    "parks": Array [],
    "searchVisible": true,
    "searching": false,
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_currentlyRenderingComponent": null,
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <App />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_previousComponentIdentity": null,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="App"
      >
        <TopBar
          onReset={[Function]}
          onSearch={[Function]}
        />
        <MapPanel
          center={
            Object {
              "lat": Object {
                "lat": 12,
                "lng": 13,
              },
              "lng": undefined,
            }
          }
          gMapsApiKey="AIzaSyCra5AQ9_m4TUQRk-iVIBRKoIQvNyRsc2E"
          onDestinationSelection={[Function]}
        >
          <AvailableParkList
            parks={Array []}
          />
          <DestinationMarker
            destination={
              Object {
                "lat": Object {
                  "lat": 12,
                  "lng": 13,
                },
                "lng": undefined,
              }
            }
            isLoading={false}
            onMarkerClick={[Function]}
            onSearch={[Function]}
            onSearchClosed={[Function]}
            searchVisible={true}
          />
          <SuggestedParkList
            parks={Array []}
          />
        </MapPanel>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;

exports[`<App> Should show the marker if a destination is clicked 1`] = `
App {
  "_loadAllParks": [Function],
  "_parseAvailableParkResponse": [Function],
  "_parseMeasurementResponse": [Function],
  "_showMessageDialog": [Function],
  "context": Object {},
  "onDestinationClick": [Function],
  "onDialogClosed": [Function],
  "onSearchWindowClosed": [Function],
  "props": Object {},
  "refs": Object {},
  "reset": [Function],
  "searchParks": [Function],
  "searchStreetByName": [Function],
  "setDestination": [Function],
  "setState": [Function],
  "state": Object {
    "availableParks": Array [],
    "destination": null,
    "parks": Array [],
    "searchVisible": true,
    "searching": false,
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_currentlyRenderingComponent": null,
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <App />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_previousComponentIdentity": null,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="App"
      >
        <TopBar
          onReset={[Function]}
          onSearch={[Function]}
        />
        <MapPanel
          center={
            Object {
              "lat": 37.773972,
              "lng": -122.431297,
            }
          }
          gMapsApiKey="AIzaSyCra5AQ9_m4TUQRk-iVIBRKoIQvNyRsc2E"
          onDestinationSelection={[Function]}
          zoom={13}
        >
          <AvailableParkList
            parks={Array []}
          />
          <DestinationMarker
            destination={null}
            isLoading={false}
            onMarkerClick={[Function]}
            onSearch={[Function]}
            onSearchClosed={[Function]}
            searchVisible={true}
          />
          <SuggestedParkList
            parks={Array []}
          />
        </MapPanel>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;
